### PR TITLE
make all operation states non-movable

### PR DESCRIPTION
**Pull request by ericniebler**

### CODE REVIEW

File: test/cpos/test_cpo_start.cpp [hunk 1]
```diff
@@ -16,28 +16,28 @@
...
 
-struct my_oper {
+struct my_oper : non_movable {
   bool started_{false};
-
   friend void tag_invoke(ex::start_t, my_oper& self) { self.started_ = true; }
 };
 
-struct op_value {
+struct op_value /*: non_movable*/ {
```
Comment (miscco) [inline test/cpos/test_cpo_start.cpp:28]: Are those comments intentional?

Comment (ericniebler) [reply test/cpos/test_cpo_start.cpp:28]: Yeah, the test case fails if the types is immovable. I want to remember that this one is intentionally movable, so I left it commented.

Comment (miscco) [reply test/cpos/test_cpo_start.cpp:28]: I would have preferred something along `/* intentionally movable */` but a comment is definitely better than none

Comment (ericniebler) [reply test/cpos/test_cpo_start.cpp:28]:
```suggestion
struct op_value /*: non_movable*/ { // Intentionally movable!
```

**Pull request merged**